Stresser/RuleController.cpp

```diff
@@ -8,8 +10,14 @@ RuleController& RuleController::getInstance(AuthorizedHttpRequest& authorizedHtt
 	return g_instance;
 }
 
-RuleEntity RuleController::getRule(const std::string& ruleId) const
-{
+std::unique_ptr<RuleEntity> RuleController::getRule(const std::string& ruleId) const
```

Comment:
Classic implementation for Factory Pattern.
In this solution we are violates the Single Responsibility principle.
RuleController role is to take of fetching Rule data and return the result to the user.
It should NOT know how to create RuleEntity.
Let's talk about it.